src/DAQProcess.cc

```diff
@@ -0,0 +1,47 @@
+/**
+ * @file DAQProcess class implementation
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have received with this code.
+ */
+
+#include "app-framework/DAQProcess.hh"
+
+#include "app-framework-base/Services/CommandFacility.hh"
+#include "app-framework-base/Services/ConfigurationManager.hh"
+#include "app-framework-base/Services/Logger.hh"
+#include "app-framework-base/Services/ServiceManager.hh"
+
+namespace appframework {
+ServiceManager* ServiceManager::handle_ = nullptr;
+ConfigurationManager* ConfigurationManager::handle_ = nullptr;
+
+DAQProcess::DAQProcess(std::list<std::string> args) : myState_(StateMachineState::Initial) {
+    Logger::setup(args);
+    CommandFacility::setup(args);
+    ConfigurationManager::setup(args);
+    ServiceManager::setup(args);
+    myState_ = StateMachineState::Booted;
+}
+
+void DAQProcess::register_modules(std::unique_ptr<ModuleList> const& ml) {
+    ml->ConstructGraph(bufferMap_, userModuleMap_, transitionOrderMap_);
+    myState_ = StateMachineState::ModulesRegistered;
+}
+
+void DAQProcess::execute_transition(TransitionName cmd) {
+    if (transitionOrderMap_.count(cmd)) {
+        for (auto& moduleName : transitionOrderMap_[cmd]) {
+            if (userModuleMap_.count(moduleName)) {
+                userModuleMap_[moduleName]->execute_transition(cmd);
```

Comment:
Both the execute_transision() calls should be inside a try/catch loop. Question we have to answer is: what do we do if we catch something? Probably at this point, only dying.